### PR TITLE
fix(orchestrator): reject malformed analytics pagination

### LINKED ISSUE #987: fix analytics events pagination should reject malformed numeric query strings
## Description
The analytics events route accepts malformed pagination query strings because `readPositiveInteger` uses `Number.parseInt` and treats any positive numeric prefix as valid. Requests like `GET /api/analytics/events?page=1abc&pageSize=50xyz` are parsed as page 1 / pageSize 50 instead of being rejected or ignored as malformed.

## Affected locations
- `packages/franken-orchestrator/src/http/routes/analytics-routes.ts:28-30` wires `/events` directly to `readPageRequest(c.req.query())`.
- `packages/franken-orchestrator/src/http/routes/analytics-routes.ts:43-52` forwards parsed `page` and `pageSize` to the analytics service.
- `packages/franken-orchestrator/src/http/routes/analytics-routes.ts:65-70` uses `Number.parseInt(value, 10)` and only checks `Number.isFinite(parsed) && parsed > 0`, so partial values like `2foo` are accepted as `2`.
- `packages/franken-orchestrator/src/analytics/sqlite-analytics-service.ts:150-161` trusts the parsed page request and clamps only numeric ranges, not malformed source strings.

## Reasoning / reproduction path
1. Call `/api/analytics/events?page=2abc&pageSize=25junk`.
2. `readPositiveInteger('2abc')` returns `2`; `readPositiveInteger('25junk')` returns `25`.
3. The response is indistinguishable from a correctly formed `page=2&pageSize=25` request, hiding operator typos and making pagination links/debugging misleading.

## Impact
Low-risk API validation bug. Operators and UI callers can send malformed analytics URLs and receive apparently valid pages, which makes troubleshooting analytics filters harder and weakens regression coverage for query validation.

## Recommendation
Parse pagination with a strict decimal-integer check such as `/^[1-9]\d*$/` before converting, or return a 422 validation error for malformed `page` / `pageSize` values. Add route tests for valid integers, zero/negative values, partial numeric strings, decimals, and non-numeric strings.

## Acceptance criteria
- `page=2abc`, `pageSize=25junk`, decimal, negative, and zero values are not silently treated as valid pagination controls.
- Valid positive integers continue to work and are still clamped by the service limit.
- Unit/route tests cover malformed pagination query strings.

Duplicate check: read `/home/pfkagent/.hermes/state/frankenbeast_open_issues_cache.json` and searched cache/live issues for `analytics`, `pagination`, `pageSize`, `parseInt`, and `readPositiveInteger`. Existing open issues found for CLI/MCP numeric validation (#974, #972, #983) but none for analytics pagination query parsing.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- reject malformed analytics events pagination query values instead of partially parsing numeric prefixes
- return a structured invalid_pagination error before querying analytics data
- cover numeric prefixes, decimals, zero, negative, and non-numeric pagination values

## Test Plan
- npm test --workspace @franken/orchestrator -- tests/unit/http/analytics-routes.test.ts
- npm run build --workspace @franken/types
- npm run build --workspace @franken/planner
- npm run build --workspace @franken/brain
- npm run build --workspace @franken/observer
- npm run build --workspace @franken/critique
- npm run build --workspace @franken/governor
- npm run typecheck --workspace @franken/orchestrator
- npm run lint --workspace @franken/orchestrator
- npm run build --workspace @franken/orchestrator
- npm test --workspace @franken/orchestrator

Closes #987

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: packages/franken-orchestrator/src/http/routes/analytics-routes.ts [hunk 1]
```diff
@@ -97,10 +108,25 @@ function invalidTimeWindow(c: Context) {
...
+function readPositiveInteger(value: string | undefined): PositiveIntegerResult {
+  if (value === undefined || value === '') {
+    return { ok: true };
+  }
+  if (!/^[1-9]\d*$/.test(value)) {
+    return { ok: false };
   }
-  const parsed = Number.parseInt(value, 10);
-  return Number.isFinite(parsed) && parsed > 0 ? parsed : undefined;
+  return { ok: true, value: Number(value) };
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-orchestrator/src/http/routes/analytics-routes.ts]: Fixed in the latest push: pagination values are now parsed only after the decimal-integer regex passes, and the parsed number must be `Number.isSafeInteger(parsed)`. I added regression coverage for a 400-digit `page` value and re-ran targeted route tests, orchestrator typecheck/lint/build, and the full orchestrator test suite.

**Comment by djm204** [comment]
@codex review
